flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcFilterPushdownVisitor.java

```diff
@@ -53,7 +52,6 @@
  * Visitor to walk a Expression AST. Produces a String that can be used to pushdown the filter,
  * return Optional.empty() if we cannot pushdown the filter.
  */
-@Public
```

Comment:
It is not allowed to change public interface. Please check [FLIP-196: Source API stability guarantees](https://cwiki.apache.org/confluence/display/FLINK/FLIP-196%3A+Source+API+stability+guarantees) for more information.

Reply:
Hi @JingGe, thanks for the comment.

This is a file I newly added, and in my latest commit, I marked it as PublicEvolving: https://github.com/apache/flink/pull/20140/commits/6c97da0eb0d27774ed9954fbabed7b514ac8ce02

Do you mean we cannot expose new API? 
For instance, I added this here: https://github.com/apache/flink/pull/20140/files#diff-ae60653ffe2ac890a3c1b01da41405bcc4e6913949176c36edc009df5090c38fR157, which adds a new method and a new return type to `JdbcDialect` which is PublicEvolving, because filter pushdown might differ across JDBC dialect. Is this approach a problem?

Reply:
Hi @JingGe did you get a chance to check on this?

Reply:
Hi @qingwei91 Could you please explain why is a public API required? Does @Internal work too? Thanks.

Reply:
Hi @JingGe , I believe it has to be PublicEvolving because this class is returned by a public method in JdbcDialect, which is a PublicEvolving class.

I will sort out the commit, sorry for the confusion

Reply:
I've squash the commits now and we have a green build.

Can anyone of you review this again? Thank you!